DataFormats/Detectors/EMCAL/src/Cell.cxx

```diff
@@ -23,176 +23,44 @@ Cell::Cell(Short_t tower, Double_t energy, Double_t time, ChannelType_t ctype)
   setType(ctype);
 }
 
-void Cell::setTower(Short_t tower)
-{
-  if (tower > 0x7fff || tower < 0)
-    tower = 0x7fff;
-  ULong_t t = (ULong_t)tower;
-
-  ULong_t b = getLong() & 0xffffff8000; // 1111111111111111111111111000000000000000
-  mBits = b + t;
-}
-
-Short_t Cell::getTower() const
-{
-  ULong_t t = getLong();
-  t &= 0x7fff;
-  return ((Short_t)t);
-}
-
 void Cell::setTimeStamp(Double_t time)
```

Comment:
again, why double?

Reply:
In principle in order to avoid implicit type conversion (raw fitter will provide the time in double format). To be decided.

Reply:
I don't think you need double at all, in the end, you fit everyting to <=15 bits.